lib/topics_filter.rb

```diff
@@ -84,6 +86,15 @@ def filter_status(status:, category_id: nil)
 
   private
 
+  def filter_state(state:)
+    case state
+    when "pinned"
+      @scope.where("topics.pinned_at IS NOT NULL AND topics.pinned_until > topics.pinned_at")
```

Comment:
Could please clarify that logic to me? I think it should be something like `Time.now < topics.pinned_until AND Time.now > topics.pinned_at`

Reply:
The logic for `in:pinned` should be to filter for topics that are currently pinned so you're right that I have to add a `Time.zone.now < topics.pinned_until` clause in the filter.